test/qa/lib/dns.py

```diff
@@ -1,45 +1,47 @@
-from lib import logging
-import sh
+import dns.resolver
 
-# returns True when NordVPN DNS servers are used
-def _is_set_to_nord():
-    return "103.86.96.100" and "103.86.99.100" in sh.cat("/etc/resolv.conf")
 
+# Used for test parametrization.
+DNS_NORD = ["103.86.96.100", "103.86.99.100"]
+DNS_NORD_IPV6 = ["2400:bb40:4444::100", "2400:bb40:8888::100"]
 
-# returns True when Threat Protection Lite DNS servers are used
-def _is_set_to_threat_protection_lite():
-    return "103.86.96.96" and "103.86.99.99" in sh.cat("/etc/resolv.conf")
+# Used for test parametrization.
+DNS_TPL = ["103.86.96.96", "103.86.99.99"]
+DNS_TPL_IPV6 = ["2400:bb40:4444::103", "2400:bb40:8888::103"]
 
+# Used for test parametrization, when the same test has to be run for different values of custom dns.
+DNS_CASE_CUSTOM_SINGLE = "2.0.0.0"
+DNS_CASE_CUSTOM_DOUBLE = "2.0.0.1 2.0.0.2"
+DNS_CASE_CUSTOM_TRIPLE = "2.0.0.3 2.0.0.4 2.0.0.5"
+DNS_CASES_CUSTOM = [DNS_CASE_CUSTOM_SINGLE, DNS_CASE_CUSTOM_DOUBLE, DNS_CASE_CUSTOM_TRIPLE]
 
-# returns True when NordVPN DNS servers are used
-def _is_set_to_nord_ipv6():
-    return "2400:bb40:8888::100" and "2400:bb40:4444::100" in sh.cat("/etc/resolv.conf")
+ALL_DNS_CASES = \
```

Comment:
If I understand correctly how this is used, I think it should be called `ALL_TEST_DNS_ADDRESSES` instead. It's just that originally I thought this is also used in parametrization, but looking at code I see that this is used to check whether dns is unset.

Reply:
It is also used for parametrization in `test_dns.py` module, in test `test_custom_dns_connect`